src/main/java/mate/academy/dao/BookDaoImpl.java

```diff
@@ -0,0 +1,158 @@
+package mate.academy.dao;
+
+import mate.academy.ConnectionUtil;
+import mate.academy.exception.DataProcessingException;
+import mate.academy.model.Book;
+
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.FileReader;
+import java.io.IOException;
+import java.math.BigDecimal;
+import java.sql.Connection;
+import java.sql.DatabaseMetaData;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+@mate.academy.lib.Dao
+public class BookDaoImpl implements BookDao {
+    private static final String TABLE_NAME = "books";
+    private static final String TITLE_NAME = "title";
+    private static final String PRICE_NAME = "price";
+    private static final int FIRST_INDEX_SQL_DATA = 1;
+    private static final int SECOND_INDEX_SQL_DATA = 2;
+    private static final int THIRD_INDEX_SQL_DATA = 3;
+    private static final int ROW_EXIST_NUM = 1;
+
+    @Override
+    public Book create(Book book) {
+        String sql = "INSERT INTO " + TABLE_NAME + " (title, price) VALUES (?, ?)";
+        try (Connection connection = ConnectionUtil.getConnection();
+             PreparedStatement statement = connection.prepareStatement(sql, Statement.RETURN_GENERATED_KEYS)) {
+            validateTableConnection(connection, TABLE_NAME);
+            statement.setString(FIRST_INDEX_SQL_DATA, book.getTitle());
+            statement.setBigDecimal(SECOND_INDEX_SQL_DATA, book.getPrice());
+            int affectedRows = statement.executeUpdate();
+            if (affectedRows < ROW_EXIST_NUM) {
+                throw new DataProcessingException("Expected to insert at leas one row, but inserted 0 row.");
+            }
+            ResultSet generatedKeys = statement.getGeneratedKeys();
+            if (generatedKeys.next()) {
+                Long id = generatedKeys.getObject(FIRST_INDEX_SQL_DATA, Long.class);
+                book.setId(id);
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't add new book: " + book, e);
+        }
+        return book;
+    }
+
+    @Override
+    public Optional<Book> findById(Long id) {
+        String sql = "SELECT * FROM " + TABLE_NAME + " WHERE id = ?";
+        try (Connection connection = ConnectionUtil.getConnection();
+             PreparedStatement statement = connection.prepareStatement(sql)) {
+            validateTableConnection(connection, TABLE_NAME);
+            statement.setLong(FIRST_INDEX_SQL_DATA, id);
+            ResultSet resultSet = statement.executeQuery();
+            if (resultSet.next()) {
+                String title = resultSet.getString(TITLE_NAME);

```

Comment:
Do not duplicate the code. Extract mapping logic in separate private method which accepts ResultSet and use it in `finaAll()` too.